cmd/credentialUtil.go

```diff
@@ -139,17 +139,28 @@ func getBlobCredentialType(ctx context.Context, blobResourceURL string, canBePub
 // 1. If there is cached session OAuth token, indicating using token credential.
 // 2. If there is OAuth token info passed from env var, indicating using token credential. (Note: this is only for testing)
 // 3. Otherwise use shared key.
-func getBlobFSCredentialType() (common.CredentialType, error) {
+func getBlobFSCredentialType(ctx context.Context, blobResourceURL string, standaloneSAS bool) (common.CredentialType, error) {
+	resourceURL, err := url.Parse(blobResourceURL)
+	if err != nil {
+		return common.ECredentialType.Unknown(), err
+	}
+
 	if oAuthTokenExists() {
 		return common.ECredentialType.OAuthToken(), nil
 	}
 
+	sas := azbfs.NewBfsURLParts(*resourceURL).SAS
+
+	if isSASExisted := sas.Signature() != ""; isSASExisted || standaloneSAS {
+		return common.ECredentialType.Anonymous(), nil
+	}
+
 	name := os.Getenv("ACCOUNT_NAME")
 	key := os.Getenv("ACCOUNT_KEY")
 	if name != "" && key != "" { // TODO: To remove, use for internal testing, SharedKey should not be supported from commandline
 		return common.ECredentialType.SharedKey(), nil
 	} else {
-		return common.ECredentialType.Unknown(), errors.New("OAuth token or shared key should be provided for Blob FS")
+		return common.ECredentialType.Unknown(), errors.New("OAuth token or shared key should be provided for Blob FS (gbfsct)")
```

Comment:
This looks like a typo at the end of this line. (Or else a long acronym that I'm not familiar with). Also, the main body of the message is no longer correct. It should cover all three options that are now supporter, OAuth token, SAS Url, or shared key.

Reply:
Gah! You caught a leftover debug message again. gbfsct stood for getBlobFSCredentialType and it was to indicate which of the messages I was actually receiving the response from.

I've found myself doing these kinds of things less often since I started, but, it's still a bad habit to debug this way for me.